src/cr/cube/dimension.py

```diff
@@ -958,3 +958,33 @@ def label(self):
         """str display name for this subtotal, suitable for use as label."""
         name = self._subtotal_dict.get("name")
         return name if name else ""
+
+    @lazyproperty
+    def subtrahend_ids(self):
+        """tuple of int ids of elements of negative part of the subtotal.
+
+        Any element id not present in the dimension or present but
+        representing missing data is excluded.
+        """
+        return tuple(
+            arg
+            for arg in self._subtotal_dict.get("kwargs", {}).get("negative", [])
+            if arg in self._valid_elements.element_ids
+        )
+
+    @lazyproperty
+    def subtrahend_idxs(self):
+        """ndarray of int base-element offsets contributing to the negative part of subtotal.
+
+        Suitable for directly indexing a numpy array object (such as base values or
+        margin) to extract the addend values for this subtotal.
+        """
+        subtrahend_ids = self.subtrahend_ids
+        return np.fromiter(
+            (
+                idx
+                for idx, vector in enumerate(self._valid_elements)
+                if vector.element_id in subtrahend_ids
+            ),
+            dtype=int,
+        )
```

Comment:
`subtrahend_ids` is only used once. Why wouldn't you inline `self.subtrahend_ids`?

Might also be worth a comment mentioning that the reason the `dtype=int` is important is that this value is used for indexing, an empty array takes the default type `float` and an array of type float cannot be used for indexing, so this is needed to prevent an error when the array is empty (as it often is).